Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM golang:1.19.1-alpine3.16 AS builder
+
+WORKDIR /app
+RUN set -eux; \
+    apk add --no-cache \
+        gc-dev \
+        make \
+        gcc \
+        git \
+        musl-dev \
+        pcre-dev \
+    ;
+
+COPY . .
+RUN make
+
+
+
+FROM alpine:3.16 AS production
```

Comment:
nit: the `production` name isn't too friendly to the 12 factor mindset. How about not giving any name to it?